integration/querydsl/expressions/src/main/java/com/blazebit/persistence/querydsl/BlazeCriteriaBuilderRenderer.java

```diff
@@ -1457,4 +1479,72 @@ public Expression<?> visit(TemplateExpression<?> expr, BindResolverContext conte
 
     }
 
+    /**
+     * Recursively find CTE's and render these against the CriteriaBuilder.
+     *
+     * @since 1.6.7
+     */
+    private class CTERenderVisitor implements Visitor<Object, Object> {
```

Comment:
Maybe I'm not seeing it, but can't this class be static?

Reply:
I don't think so, it needs to invoke some methods on the outer renderer.